zipline/finance/performance/period.py

```diff
@@ -203,9 +200,12 @@ def calculate_performance(self):
 
     def record_order(self, order):
         if self.keep_orders:
-            dt_orders = self.orders_by_modified[order.dt]
-            if order.id in dt_orders:
-                del dt_orders[order.id]
+            try:
+                dt_orders = self.orders_by_modified[order.dt]
+                if order.id in dt_orders:
+                    del dt_orders[order.id]
```

Comment:
Oh, this still does use an OrderedDict. So should I think we should preserve it, though that behavior appears to be uncovered in the zipline tests. (Was able to remove the `del` and tests still passed.)